index.js

```diff
@@ -18,27 +18,35 @@ var FILTER = filterize([
   /^\._/,
 ]);
 
-function pack(folder) {
-  var tarPack = tar.Pack({ noProprietary: true });
-  var contents = fstream.Reader({
-    path: folder,
-    type: 'Directory',
-    isDirectory: true,
-    filter: FILTER,
+function pack(folder, opts) {
+  opts = opts || {};
+  opts.injections = opts.injections || [];
+  var tarPack = tar.pack(folder, {
+    ignore: FILTER,
+    map: function(header) {
+      if (header.name === '.') {
+        header.name = 'package';
+      } else {
+        header.name = 'package/' + header.name;
+      }
+      return header;
+    },
   });
   if (debug.enabled) {
-    contents.on('error', debug.bind(null, 'err reading folder'));
     tarPack.on('error', debug.bind(null, 'tar creation error'));
   }
-  return contents.pipe(tarPack);
+  opts.injections.forEach(function(i) {
+    tarPack.entry({name: i.path}, i.body);
```

Comment:
Pretty much.

Basically the `tar-fs` module is awesome and provides the APIs I keep wrapping around the `tar` module.